be/src/util/threadpool.h

```diff
@@ -331,24 +334,24 @@ class ThreadPool {
     // A thread is added to the front of the list when it goes idle and is
     // removed from the front and signaled when new work arrives. This produces a
     // LIFO usage pattern that is more efficient than idling on a single
-    // ConditionVariable (which yields FIFO semantics).
     //
     // Protected by _lock.
     struct IdleThread : public boost::intrusive::list_base_hook<> {
-        explicit IdleThread(Mutex* m) : not_empty(m) {}
+        explicit IdleThread() {}
 
         // Condition variable for "queue is not empty". Waiters wake up when a new
         // task is queued.
-        ConditionVariable not_empty;
-
-        DISALLOW_COPY_AND_ASSIGN(IdleThread);
+        std::condition_variable not_empty;
+        IdleThread(const IdleThread&) = delete;
+        void operator=(const IdleThread&) = delete;
     };
     boost::intrusive::list<IdleThread> _idle_threads; // NOLINT(build/include_what_you_use)
 
     // ExecutionMode::CONCURRENT token used by the pool for tokenless submission.
     std::unique_ptr<ThreadPoolToken> _tokenless;
 
-    DISALLOW_COPY_AND_ASSIGN(ThreadPool);
```

Comment:
why not use the macros?

Reply:
DISALLOW_COPY_AND_ASSIGN  may be conflict with brpc, and I will use bvar in runtime profile in next pr